### PR TITLE
Remove python dependency by generating crypto keys using java

### PR DESCRIPTION
**Pull request by steven-gueguen**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/GenerateCryptoKey.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Arrays.asList(configFiles).forEach(file ->
+      {
+        Path p = FileSystems.getDefault().getPath(file);
+        // Read current file
+        try (Stream<String> lines = Files.lines(p)) {
+          List<String> replaced = lines
+            .map(line -> line.replaceAll("(.* = hex:)(.*)", "$1" + generateCryptoKey(256)))
+            // *.hash.* keys need shorted crypto key
+            .map(line -> line.replaceAll("(.*\\.hash\\..* = hex:)(.*)", "$1" + generateCryptoKey(128)))
+            .map(line -> line.replaceAll("(.*fetch.psk.* = hex:)(.*)", "$1" + generateCryptoKey(128)))
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/GenerateCryptoKey.java:62]: You are missing 128 bits keys, `directory.psk` for instance. Check all references to `KeyStore.checkAndSetKey` to find all used keys and their length. The cleanest (and tedious) way would be to have a Map in a similar way Python did. 

You should also compile your regexs outside this stream for performance.

**Review by ftence — CHANGES_REQUESTED** [review]

File: warp10/src/main/java/io/warp10/GenerateCryptoKey.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            .collect(Collectors.toList());
+          Files.write(p, replaced);
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (steven-gueguen) [reply warp10/src/main/java/io/warp10/GenerateCryptoKey.java:71]: done

File: warp10/src/main/java/io/warp10/GenerateCryptoKey.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        try (Stream<String> lines = Files.lines(p)) {
+          List<String> replaced = lines
+            .map(line -> line.replaceAll("(.* = hex:)(.*)", "$1" + generateCryptoKey(256)))
+            // *.hash.* keys need shorted crypto key
+            .map(line -> line.replaceAll("(.*\\.hash\\..* = hex:)(.*)", "$1" + generateCryptoKey(128)))
+            .map(line -> line.replaceAll("(.*fetch.psk.* = hex:)(.*)", "$1" + generateCryptoKey(128)))
+            .collect(Collectors.toList());
+          Files.write(p, replaced);
+        } catch (IOException e) {
+          e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (steven-gueguen) [reply warp10/src/main/java/io/warp10/GenerateCryptoKey.java:66]: done

**Comment by hbs** [comment]
What is the status of the last change requested by @ftence?

**Comment by hbs** [comment]
ping?

**Comment by ftence** [comment]
Waiting for https://github.com/steven-gueguen/warp10-platform/pull/1 to be merged or not.

**Review by pi-r-p — COMMENTED** [review]
Remarks done on @ftence  PR on @steven-gueguen  PR...

**Review by ftence — APPROVED** [review]

File: warp10/src/main/java/io/warp10/GenerateCryptoKey.java (hunk 1 above)
Comment (steven-gueguen) [reply warp10/src/main/java/io/warp10/GenerateCryptoKey.java]: Thanks for the PR

**Comment by pi-r-p** [comment]
Standalone Bootstrap test:

```
cat * | grep -B4 hex:h
//
// OSS Master Key, used to decrypt any 'wrapped:base64' key
// Valid formats are hex:... or base64:....
//
#oss.master.key = hex:hhhhhh...
--
//
// @deprecated
// 128 bits SipHash key for verifying index names. Valid formats are hex:..., base64:... or, when using OSS, wrapped:....
//
warp.hash.index = hex:hhhhhh...
--
// @deprecated
// 128/192/256 bits AES key for encrypting index details in in LevelDB.
// Valid formats are hex:..., base64:... or, when using OSS, wrapped:....
//
#leveldb.index.aes = hex:hhhhhh...
```

What about removing warp.hash.index and leveldb.index.aes from the templates ?

**Comment by hbs** [comment]
Yes they should be gone.

**Comment by ftence** [comment]
But this is outside the scope of this PR.

**Review by pi-r-p — APPROVED** [review]

**Pull request merged**